File: lua/5-loops.lua
#!/bin/lua

e=0

for i=0, 9, 1
do
	print("inc i=", i)
	b=i
end

for i=b, 20
do
	print("inc auto i=", i)
	c=i
end

for i=c, 1, -1
do
	print("dec i=", i)
	d=i
end

print("start: d=", d);
while (d < 50)
do
	print("d=", d, "e=", e)
	e=e+1
	d=d+1
	if (e > 10) 
	then
		break
	end
end

print("start: d=", d);
repeat
	print("d=", d)
	d=d+1
until d > 15

for  h=0, 10
do
	if h == 4
	then
		print("I've got 4")
		goto continue -- starting with lua 5.2
	elseif h== 5 
	then 
			print("side jump", h)
			goto continue
		elseif h == 1
		then
			print("one")
		else
			print("something other than 1, because", h)
	end
	::continue::
end

File: rust/5-loops.rs
fn main() {
    let mut b=0;
    let mut c;
    let mut d;
    let mut e;
    let mut f;
    let mut g = 0;

    for i in 0..10 {
        println!("i={}", i);
        b = i;
    }

    println!("start: b=i={}", b);
    while b < 20 {
        println!("b++={}", b);
        b += 1;
    }

    println!("start: c=b={}", b);
    c = b;
    while c < 30 {
        c += 1;
        println!("++c={}", c);
    }

    println!("start: d=c, e=c={}", c);
    d = c;
    e = c;
    while d < 40 {
        d += 1;
        e += 1;
        println!("++d={} e++={}", d, e);
    }

    f = e;
    println!("start: f=e={}", e);
    while e < 50 {
        e += 1;
        e -= 1;
        f -= 1;
        f += 1;
        println!("++e={} e--={} f--={} ++f={}", e, e, f, f);
        g += 1;
        if g > 10 {
            break;
        }
    }

    println!("start: g={}", g);
    while g > 0 {
        println!("g--={}", g);
        g -= 1;
    }

    for h in 0..10 {
        if h == 4 {
            println!("I've got 4");
            continue;
        }
        if h == 5 {
            println!("side jump: {}", h);
            continue;
        }
        match h {
            1 => println!("one"),
            _ => println!("something other than 1, because {}", h),
        }
    }
}
